corehq/apps/app_manager/models.py

```diff
@@ -3469,6 +3469,21 @@ def set_media_versions(self, previous_version):
             else:
                 map_item.version = self.version
 
+    def ensure_module_unique_ids(self, should_save=False):
+        """
+            Creates unique_ids for modules that don't have unique_id attributes
+            should_save: the doc will be saved only if should_save is set to True
+
+            WARNING: If called on the same doc in different requests without saving,
+            this function will set different uuid each time,
+            likely causing unexpected behavior
+        """
+        if any(not mod.unique_id for mod in self.modules):
+            for mod in self.modules:
+                mod.get_or_create_unique_id()
+            if should_save:
+                self.save()
```

Comment:
I would prefer an approach here that only saves if it actually needs to make any changes. So rather then calling `get_or_create_unique_id` on all module if any of them are missing just check each module in the loop and only call the method if the unique ID is missing for that module. Also keeping track of if you've made any changes and only save if you have.